sympy/ntheory/residue_ntheory.py

```diff
@@ -151,6 +151,31 @@ def primitive_root(p):
     return next(_primitive_root_prime_iter(p))
 
 
+def factmod(n, p):
```

Comment:
What will happen if `n` or `p` or both are symbols?

Reply:
Maybe I should also insert these two lines of code:
```
n = as_int(n)
p = as_int(p)
```

Reply:
I think the result should remain unevaluated if symbols are passed in the argument.

Reply:
How should I handle this now?

Reply:
Most functions in this module are non-symbolic and are intended to work with plain integers. I have the impression that this module is used for basic numeric algorithms that can be used to build up symbolic code elsewhere.

On that basis I wouldn't expect this to handle symbols or return a symbolic unevaluated object.

I don't know what the intended direction of travel is for ntheory in general though.

Reply:
I agree. No function in `residue_ntheory` handles the case for symbols. 
@oscarbenjamin Please look into [this](https://github.com/sympy/sympy/commit/6f479c61bd5333f4a2d7631f1ae5f30eff94df31) and let me know if I can proceed further on this one.

Reply:
I would raise an another questions like:
Should it check that `n` and `p` are integers and prime, and should raise an error?

Reply:
And I would also name like `fact_mod` like how `sqrt_mod` or `nthroot_mod` is named like

Reply:
> Should it check that `n` and `p` are integers and prime

Sure, I can add those checks in the beginning. 
i'll change it to `fact_mod`.